### DISCUSSION TITLE
[Azure Pipelines] Scaling target to have one idle agent

### DISCUSSION THREAD

**Discussion by ToniA** [kedacore/keda#3510] [Ideas]
Currently the Azure DevOps Pipelines KEDA agent scales based on the queue length, i.e. starts new Azure DevOps build agents whenever there are queueing builds. This works fine, but due to our memory requirements for builds we have set the agent POD to require the whole node. So each time the agent pool size is increased, a new Kubernetes node is started. The problem is that it takes about 5-6 minutes to start a new agent. Due to this, I have set my minimum pool size to 2 agents.

I was wondering if it would make sense to provide an additional scaling target, as a number of idle build agents. This would start a new agent every time there are no idle build agent. So in the case of no running builds, you'd have the compulsory agent (minimum is always one), in case of one running build, it would immediately start a second agent etc.

**Comment by tomkerkhove** [comment]
A current workaround would be to have 2 workloads as part of the same pool:
- First workload is being autoscaled based on pending jobs with ScaledJob (as per https://keda.sh/blog/2021-05-27-azure-pipelines-scaler/)
- Second workload that is not autoscaled and pinned to 1 instance so that it acts as an idle agent.

Our ScaledObject has `idleReplicaCount` but this is not an option given it's not a fit for this workload type imo.
